test/unit/RedisResourceManagerTest.php

```diff
@@ -287,6 +309,36 @@ public function testWillCatchAuthDuringConnectException(): void
         $this->resourceManager->getResource('default');
     }
 
+    public function testWillCatchAuthDuringConnectExceptionWithUser(): void
+    {
+        $redis = $this->createMock(Redis::class);
+
+        $redis
+            ->method('connect')
+            ->willReturn(true);
+
+        $redis
+            ->expects(self::atLeastOnce())
+            ->method('auth')
+            ->with(['dummyuser', 'secret'])
+            ->willThrowException(new RedisException('test'));
```

Comment:
why would that method call throw an exception?

If you have `expects` defined so that it is verified that the method is called for sure, that is already an assertion.
By providing `with`, that would explicitly check that when the method is invoked, all arguments are matching the expectations we passed here. 

Is the test failing when you remove that exception?